scaleway/resource_baremetal_server_beta.go

```diff
@@ -0,0 +1,220 @@
+package scaleway
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	baremetal "github.com/scaleway/scaleway-sdk-go/api/baremetal/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayBaremetalServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayBaremetalServerCreate,
+		Read:   resourceScalewayBaremetalServerRead,
+		Update: resourceScalewayBaremetalServerUpdate,
+		Delete: resourceScalewayBaremetalServerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Name of the server.",
+			},
+			"type": {
```

Comment:
I'm not sure we shout change the name of the API here. This is a beta resource hopefully API will change before GA.